x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/string/TopSnippetsTests.java

```diff
@@ -0,0 +1,253 @@
+import org.elasticsearch.xpack.esql.expression.function.TestCaseSupplier;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static org.elasticsearch.compute.data.BlockUtils.toJavaObject;
+import static org.elasticsearch.xpack.esql.expression.function.scalar.string.TopSnippets.DEFAULT_NUM_SNIPPETS;
+import static org.elasticsearch.xpack.esql.expression.function.scalar.string.TopSnippets.DEFAULT_WORD_SIZE;
+import static org.elasticsearch.xpack.esql.expression.function.scalar.util.ChunkUtils.chunkText;
+import static org.hamcrest.Matchers.equalTo;
+
+public class TopSnippetsTests extends AbstractScalarFunctionTestCase {
+
+    private static String PARAGRAPH_INPUT = """
+        The Adirondacks, a vast mountain region in northern New York, offer a breathtaking mix of rugged wilderness, serene lakes,
+        and charming small towns. Spanning over six million acres, the Adirondack Park is larger than Yellowstone, Yosemite, and the
+        Grand Canyon combined, yet it's dotted with communities where people live, work, and play amidst nature. Visitors come year-round
+        to hike High Peaks trails, paddle across mirror-like waters, or ski through snow-covered forests. The area's pristine beauty,
+        rich history, and commitment to conservation create a unique balance between wild preservation and human presence, making
+        the Adirondacks a timeless escape into the tranquility of nature.
+        """;
+
+    public TopSnippetsTests(@Name("TestCase") Supplier<TestCaseSupplier.TestCase> testCaseSupplier) {
+        this.testCase = testCaseSupplier.get();
+    }
+
+    private static String randomWordsBetween(int min, int max) {
+        return IntStream.range(0, randomIntBetween(min, max))
+            .mapToObj(i -> randomAlphaOfLengthBetween(1, 10))
+            .collect(Collectors.joining(" "));
+    }
+
+    @ParametersFactory
+    public static Iterable<Object[]> parameters() {
+        return parameterSuppliersFromTypedDataWithDefaultChecks(
+            true,
+            List.of(new TestCaseSupplier("TopSnippets with defaults", List.of(DataType.KEYWORD, DataType.KEYWORD), () -> {
+                String text = randomWordsBetween(25, 50);
+                String query = randomFrom("park", "nature", "trail");
+                ChunkingSettings chunkingSettings = new SentenceBoundaryChunkingSettings(DEFAULT_WORD_SIZE, 0);
+
+                List<String> chunks = chunkText(text, chunkingSettings);
+                MemoryIndexChunkScorer scorer = new MemoryIndexChunkScorer();
+                List<String> scoredChunks = scorer.scoreChunks(chunks, query, DEFAULT_NUM_SNIPPETS, false)
+                    .stream()
+                    .map(MemoryIndexChunkScorer.ScoredChunk::content)
+                    .limit(DEFAULT_NUM_SNIPPETS)
```

Comment:
Curious, do we need to limit if we're specifying the num of snippets to return in scoreChunks?

Reply:
You are correct, that's a copy-paste error, thanks for catching!